tests/integration/Filters.test.js

```diff
@@ -0,0 +1,331 @@
+import Filters from '@/components/Filters';
+import Restaurant from '@/views/Restaurant';
+
+// Data
+import dishes from '@/assets/data/dishes.json';
+import restaurants from '@/assets/data/restaurants.json';
+
+import Vuetify from 'vuetify';
+import { vuetifyConfig } from '@/plugins/vuetifyConfig';
+import { createLocalVue, mount } from '@vue/test-utils';
+import { cloneDeep } from 'lodash';
+import Vuex from 'vuex';
+import storeConfig from '@/store/store';
+
+jest.mock('@/services/restaurantService');
+jest.mock('@/services/dishService');
+jest.mock('@/services/courseService');
+const restaurantService = require('@/services/restaurantService');
+const dishService = require('@/services/dishService');
+const courseService = require('@/services/courseService');
+
+describe("Filters.vue", () => {
+
+    let vuetify, wrapper, store;
+
+    const localVue = createLocalVue();
+    localVue.use(Vuex);
+
+    afterEach(() => {
+        if(wrapper){
+            wrapper.destroy();
+        }
+        vuetify = undefined;
+    });
+
+    it("pop up appears on button click", async () => {
+
+        wrapper = await mountMenu(dishes);
+
+        let filterButton = wrapper.find("button.mx-2");
+
+        expect(filterButton.exists()).toBeTruthy();
+        expect(filterButton.find("i.mdi-filter")).toBeTruthy();
```

Comment:
I don't think there is a need to ensure the icon exists. The only think that matters is that it switches correctly after filtering or filter clearing

We don't need to check if the button exists either because we a clicking it. If it doesn't exist, there will be an error and the test will fail

Reply:
Resolved

Reply:
Let's talk about at Tuesday Meeting at 28/12/2021 3 PM